modules/core/src/test/java/org/apache/ignite/failure/FailureHandlerLoggingTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.failure;
+
+import java.util.Collections;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.testframework.ListeningTestLogger;
+import org.apache.ignite.testframework.LogListener;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static org.apache.ignite.internal.processors.failure.FailureProcessor.FAILURE_LOG_MSG;
+import static org.apache.ignite.internal.processors.failure.FailureProcessor.IGNORED_FAILURE_LOG_MSG;
+
+/**
+ * Tests log messages of failure handler.
+ */
+public class FailureHandlerLoggingTest extends GridCommonAbstractTest {
+    /** Logger for listen messages. */
+    private static final ListeningTestLogger TEST_LOG = new ListeningTestLogger(false, GridCommonAbstractTest.log);
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg =  super.getConfiguration(igniteInstanceName);
+
+        TestFailureHandler hnd = new TestFailureHandler(false);
+
+        hnd.setIgnoredFailureTypes(Collections.singleton(FailureType.SYSTEM_WORKER_TERMINATION));
+
+        cfg.setFailureHandler(hnd);
+
+        cfg.setGridLogger(TEST_LOG);
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
+        startGrid(0);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTestsStopped() throws Exception {
+        super.afterTestsStopped();
+
+        stopAllGrids();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+
+        TEST_LOG.clearListeners();
+    }
+
+    /**
+     * Tests log message for ingnored failure types.
+     */
+    @Test
+    public void testFailureHandlerLoggedIgnoredFailureTest() throws Exception {
```

Comment:
Now you have the same code for both methods. Only two differences: failure type and exepcted messages. Could be moved to one method.

Reply:
According to test plan there should be two different tests https://ggsystems.atlassian.net/browse/GG-25912